core/Pos/Binary/Class.hs

```diff
@@ -218,7 +218,9 @@ getTinyVarInt' = do
     if testBit a 7
         then do
             b <- getWord8
-            if | testBit b 7 -> fail "getTinyVarInt': more than 2 bytes"
+            emptyBuf <- isEmpty
```

Comment:
@neongreen if you don't think there's a problem with these lines I'll keep them here.